contrib/node/src/python/pants/contrib/node/tasks/node_resolve.py

```diff
@@ -66,23 +72,58 @@ def _can_resolve_target(self, target):
     """
     return self.is_node_package(target) and self._resolver_for_target(target) != None
 
+  def _do_local_invalidation_check(self,
+                                   targets,
+                                   invalidate_dependents=True,
+                                   topological_order=True):
+    """Returns an invalidation_check with all targets invalidated.
+    """
+    # Generate a cache_key that never matches to ensure work is always done.
```

Comment:
If you're going to do this, then you don't actually need to use an invalidation check.

This usage of the invalidation check is incorrect anyway though I think... it's meant to be used as context manager, such that when the context exits, targets are marked valid.

Reply:
That's true, I admit this is a hack job to re-use topological sort for targets. I can copy-paste the topological sort and it should make more sense.